trlx/ray_train/accelerate_trainer.py

```diff
@@ -169,7 +169,7 @@ def wrapped_train_loop_per_worker(*args, **kwargs):
                 os.environ["LOCAL_RANK"] = str(session.get_local_rank())
                 os.environ["LOCAL_WORLD_SIZE"] = str(session.get_local_world_size())
                 os.environ["LOCAL_SIZE"] = str(session.get_local_world_size())
-                os.environ["ACCELERATE_TORCH_DEVICE"] = str(get_device())
+                os.environ["ACCELERATE_TORCH_DEVICE"] = f"cuda:{session.get_local_rank()}"
```

Comment:
qq, what's the reason for this change?

Reply:
this is a crude fix for device mismatch errors which happen otherwise

Reply:
Are you manually setting the devices anywhere else?

Reply:
Interestingly we do not, however with `get_device` there are always some discrepancies
as for example with deepspeed: 
```
os.environ.get("LOCAL_RANK", 0)='4' 
self.accelerator.device=device(type='cuda', index=3)
next(self.model.parameters()).device=device(type='cuda', index=4)
```

and even with ddp: 
```
os.environ.get("LOCAL_RANK", 0)='1'
batch.input_ids.device=device(type='cuda', index=7)
self.accelerator.device=device(type='cuda', index=7)
next(self.model.parameters()).device=device(type='cuda', index=7)
```
there is still an error even though batch is deliberately transfered https://github.com/CarperAI/trlx/blob/e72f7d1a8008c9a994e9fe465aa4a8a7a1fb3232/trlx/trainer/accelerate_ilql_trainer.py#L57 

```
File "/trlx/examples/ilql_sentiments.py", line 36, in main
  trlx.train(
File "/trlx/trlx/trlx.py", line 119, in train
  trainer.learn()
File "/trlx/trlx/trainer/accelerate_base_trainer.py", line 477, in learn
  loss, stats = self.loss(batch)
File "/trlx/trlx/trainer/accelerate_ilql_trainer.py", line 55, in loss
  logits, qs, target_qs, vs, _ = self.model(
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
  return forward_call(*input, **kwargs)
File "/trlx/.env/lib/python3.9/site-packages/accelerate/utils/operations.py", line 489, in __call__
  return convert_to_fp32(self.model_forward(*args, **kwargs))
File "/trlx/.env/lib/python3.9/site-packages/torch/amp/autocast_mode.py", line 14, in decorate_autocast
  return func(*args, **kwargs)
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1040, in forward
  output = self._run_ddp_forward(*inputs, **kwargs)
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/parallel/distributed.py", line 1000, in _run_ddp_forward
  return module_to_run(*inputs[0], **kwargs[0])
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
  return forward_call(*input, **kwargs)
File "/trlx/trlx/trainer/nn/ilql_models.py", line 249, in forward
  out = self.base_model.transformer(**forward_kwargs)
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
  return forward_call(*input, **kwargs)
File "/trlx/.env/lib/python3.9/site-packages/transformers/models/gpt2/modeling_gpt2.py", line 830, in forward
  inputs_embeds = self.wte(input_ids)
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
  return forward_call(*input, **kwargs)
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/modules/sparse.py", line 160, in forward
  return F.embedding(
File "/trlx/.env/lib/python3.9/site-packages/torch/nn/functional.py", line 2210, in embedding
  return torch.embedding(weight, input, padding_idx, scale_grad_by_freq, sparse)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:7 and cuda:1! (when checking argument for argume
nt index in method wrapper__index_select)
```

Reply:
I see, thanks. How can I reproduce this?

Reply:
@reciprocated I'm not familiar with the Ray session API but I'm curious if this might have an effect on multi-node sweeps - have you noticed any changes there? (probably shouldn't but, again, just curious 😄)

Reply:
It has been designed to work seamlessly with multi-node setups! There should be no issues there.

One thing that is concerning is that device situation, though. Ideally we would delegate picking the device to Ray Train (which is what setting that env var achieved). I'll have to see how I can reproduce this to get to the bottom of the mismatch issue.

Reply:
@Yard1 you can reproduce with:
```
git checkout test-device-mismatch
python -m trlx.sweep -y --config configs/sweeps/ppo_sweep.yml --default_config configs/ppo_config.yml --accelerate_config configs/accelerate/zero2-bf16.yaml --num_gpus 4 examples/ppo_sentiments.py
```

@jon-tow no, I hadn't seen anything abnormal when running multi-node